### ISSUE TITLE
Internationalisation / Localisation

### ISSUE THREAD

**Issue by rengolin** [microsoft/verona#182]
We need an infrastructure to localise user interactions (error messages, input, output, etc) that spans the whole project. The first user will be the `verona-mlir` tool, but the idea is that we have a single mechanism to locale-sensitive behaviour for all tools.

The main deliverables could be:
* A set of directories, one per language, with a list of files (one per tool/namespace, whatever) with pairs of "message-id" and "translated sentence", with formatted output (using %s, %d, etc).
* An easy way to distribute the translation files (via "make install", so packagers can easily move to the right area) and enable future translations to happen without needing to recompile the project.
* Change the language upon command-line arguments (ex. --lang pt_BR).
* Auto-detect locale and change localisation options based on that.
* Tools to verify translation files sanity, ex. that the "base" language has same or more "message-id"s than any other language, that equal "message-id"s in different languages have the exact same number of format options, etc.
* Not enforce any specific base language (ex. English), but allow projects to choose the base language either statically or dynamically.

**Comment by rengolin** [comment]
Work in progress: https://github.com/rengolin/verona/commits/i18n

**Comment by rengolin** [comment]
I have made an initial implementation of the internationalisation effort. It's by very far not complete, or even considering most points of the problem, but it's a start.

However, given that we're still developing a prototype that will be used to bootstrap the actual compiler, it feels as if spending time localising the code is not worthy. Even to the point of just requiring complex formatting of text will hinder efforts better spent elsewhere, in a project that will likely not last long enough to be used outside of this prototype.

So, I'm closing the issue now, but will keep the branch. If anyone feels interested in following this up, feel free to continue where I stopped.
